### ISSUE TITLE
expirevar not working; DOS blocked IP doesn't get unblocked

### ISSUE THREAD

**Issue by vojch** [coreruleset/modsecurity-crs-docker#59]
We are experiencing issues with expiring DOS block flag of IP addresses after testing an attack.

With following configuration, used for testing purposes, IP address is correctly blocked, but it doesn't get unblocked after 30s, or actually ever, unless the container is restarted.

```
SecAction \
 "id:900700,\
  phase:1,\
  nolog,\
  pass,\
  t:none,\
  setvar:'TX.DOS_BURST_TIME_SLICE=60',\
  setvar:'TX.DOS_COUNTER_THRESHOLD=20',\
  setvar:'TX.DOS_BLOCK_TIMEOUT=30'"
```

```
Docker image used: owasp/modsecurity-crs:nginx
Image ID: 253082e388a8
```

Doing some research we came across the issue of `expirevar`, used in REQUEST-912-DOS-PROTECTION.conf, which is not working on libModSecurity3 with Nginx.

We got it working by using a workaround configuration from this [issue comment](https://github.com/SpiderLabs/ModSecurity/issues/1803#issuecomment-474884652) and mounting it to the container instead of the default REQUEST-912-DOS-PROTECTION.conf file.

`expirevar` is also present in other rules, such as REQUEST-910-IP-REPUTATION.conf and REQUEST-913-SCANNER-DETECTION.conf.

Could this workaround (or other solution) be included in default rules in docker image?

Thanks!

**Comment by fzipi** [comment]
Sure, will take a look.

**Comment by fzipi** [comment]
@theseion Do you think the proposed workaround can be included in the nginx builds?

**Comment by theseion** [comment]
I don't see why not. However, it's not a general purpose fix for `expirevar`, only for DOS blocking, so other uses of `expirevar` will require additional work. We should discuss this in the meeting. I'll put it on the agenda.

**Comment by theseion** [comment]
So the plan is to move the DOS rules into a plugin for the v4 release. After that is done we will add the workaround for NGiNX, possibly by adding them as another plugin. That would solve your issue I believe, @vojch.

I've opened a [separate issue](https://github.com/coreruleset/coreruleset/issues/2389) to track that.
